### PR TITLE
rework the config file

### PR DESCRIPTION
**Pull request by MFraters**
Rework the config file by adding pre-processor version numbers and making it header only. The main thing was to add the pre-processor statements because that will help aspect, but I think making it header only makes more sense. 

In the future (with c++20) it should be able to become a constexpr, but for now const works fine.

### CODE REVIEW

File: include/world_builder/config.h.in [hunk 1]
```diff
@@ -23,25 +23,32 @@
...
-    static const std::string GIT_DATE;
-    static const std::string GIT_COMMIT_SUBJECT;
-  };
-
-  struct Data
+    // with c++20, this should be able to become a constexpr.
+    const std::string MAJOR = "@WORLD_BUILDER_VERSION_MAJOR@";
+    const std::string MINOR = "@WORLD_BUILDER_VERSION_MINOR@";
+    const std::string PATCH = "@WORLD_BUILDER_VERSION_PATCH@";
+    const std::string LABEL = "@WORLD_BUILDER_VERSION_LABEL@";
```
Comment (bangerth) [inline include/world_builder/config.h.in:39]: You can make them `constexpr char MAJOR[]` and that should work even with previous language standards.

**Comment by bangerth** [comment]
Looks reasonable to me.

File: include/world_builder/config.h.in (hunk 1 above)
Comment (MFraters) [reply include/world_builder/config.h.in:39]: Yes, I tried that, but then I run into issues where using `Version::MAJOR + "."`, would not work anymore, so a lot of them would need to change to something like `std::string(Version::MAJOR) + "."`. const string is probably good enough for now, but we can always change it later.

**Comment by MFraters** [comment]
@bangerth Thanks for the review!

**Pull request merged**